### PR TITLE
3720/bug/manage solr logs

### LINKED ISSUE #3720: Truncate solr service + tomcat7 logs
The logs for solr can fill up the system; the appear to have no limit

### Proposal & Constraints
- Add some options to the docker-compose to limit them: https://stackoverflow.com/questions/42510002/how-to-clear-the-logs-properly-for-a-docker-container
- Figure out how to make tomcat7 less noisy

### PR DESCRIPTION
**Pull request by jamesachamp**
Closes #3720 

Creates daily cron job in olsolr that removes all files in `/var/log/tomcat7/` that are more than seven days old.  Also sets limits on the size of Docker's olsolr logs.

### CODE REVIEW

File: conf/solr/rmlog [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
```
Comment (cdrini) [inline conf/solr/rmlog:1]: We like to add .sh to scripts: `rmlog.sh`

File: docker/Dockerfile.olsolr [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
     ln -s /var/log/tomcat7/ /usr/share/tomcat7/logs && \
     ln -s /etc/tomcat7/ /usr/share/tomcat7/conf
 
 COPY conf/solr/conf/* /etc/solr/conf/
 COPY conf/solr/haproxy.cfg /etc/haproxy/
 COPY conf/solr/start.sh /start.sh
 
+# Log management:
+COPY conf/solr/rmlog /etc/cron.daily/rmlog
+RUN chmod 755 /etc/cron.daily/rmlog
```
Comment (cdrini) [inline docker/Dockerfile.olsolr:14]: We shouldn't need to make this executable from here; try running the chmod command outside of the docker image: `chmod +x conf/solr/rmlog.sh`

**Review by cdrini — CHANGES_REQUESTED** [review]
This is actually a super elegant & simple solution! I like it :) A few small comments

File: conf/solr/rmlog (hunk 1 above)
Comment (jamesachamp) [reply conf/solr/rmlog]: Done.

File: docker/Dockerfile.olsolr (hunk 2 above)
Comment (jamesachamp) [reply docker/Dockerfile.olsolr]: That's good to know, and it worked perfectly.

File: conf/solr/rmlog.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+find /var/log/tomcat7/* -mtime 7 -exec rm {} \;
```
Comment (cdrini) [inline conf/solr/rmlog.sh:2]: I tried this locally and looks like a plus was needed here? For reasons not entirely clear to me :P 

```suggestion
find /var/log/tomcat7/* -mtime +7 -exec rm {} \;
```

**Review by cdrini — APPROVED** [review]
Lgtm! I tested running the command inside the solr container manually, and it wasn't matching any files, but a bit of _positive thinking_ seemed to fix that :) Haven't tested the full flow (i.e. cron is successfully running, etc); we're making loads of changes to the docker-compose file right now because of the py3 stuff, so would rather we get this merged in now :) I'll deploy to ol-solr1 and check tomorrow to see if it was deleted!

File: conf/solr/rmlog.sh (hunk 3 above)
Comment (jamesachamp) [reply conf/solr/rmlog.sh]: It looks like `-mtime 7` matches files that are exactly 7 days old, while `-mtime +7` matches those that are greater than seven.  More details can be found [here](https://stackoverflow.com/questions/25599094/explaining-the-find-mtime-command#answer-25599430).

**Pull request merged**